lib/active_model/serializer/adapter.rb

```diff
@@ -21,9 +21,11 @@ def serializable_hash(options = nil)
       end
 
       def as_json(options = nil)
-        hash = serializable_hash(options)
-        include_meta(hash) unless self.class == FlattenJson
-        hash
+        serializable_hash(options).tap do |hash|
```

Comment:
@bf4 I couldn't agree more with you about this approach. I will be happy to work on it, but in a different PR.